localstack/services/cloudformation/models/apigateway.py

```diff
@@ -57,6 +58,14 @@ def fetch_state(self, stack_name, resources):
         result = [r for r in result if r.get("name") == name]
         return result[0] if result else None
 
+    @staticmethod
+    def add_defaults(resource, stack_name: str):
+        role_name = resource.get("Properties", {}).get("Name")
```

Comment:
I'm generally a huge fan of defensive programming on dicts (`.get(..., {})`), but wondering if in this case we should ensure in the runtime that `Properties` are always present, and then simply access them via `resource["Properties"]`. (in fact, we're anyway doing it on line 65..) Thoughts?